### PR TITLE
Support public template

### PR DESCRIPTION
**Pull request by 928PJY**
For now, to run the local validation, we still need to clone the template repository, but it is private, so it is not accessible for the public contributors. but the schemas are already public and organized by this definition file: https://static.docs.com/ui/latest/schemas/schema_config.json.

[AB#288440](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/288440)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6550)

### CODE REVIEW

File: src/docfx/restore/PackageResolver.cs [hunk 1]
```diff
@@ -44,9 +44,13 @@ public bool TryResolvePackage(PackagePath package, PackageFetchOptions options,
             }
         }
 
-        public Package ResolveAsPackage(PackagePath package, PackageFetchOptions options)
+        public Package ResolveAsPackage(PackagePath package, PackageFetchOptions options, FileResolver fileResolver)
```
Comment (yufeih) [inline src/docfx/restore/PackageResolver.cs:47]: consider putting `FileResolver` as a member, if there is circular dependency, use `Lazy<FileResolver>`
```suggestion
        public Package ResolveAsPackage(PackagePath package, PackageFetchOptions options)
```

File: docs/specs/schema.yml [hunk 2]
```diff
@@ -485,3 +485,27 @@ outputs:
...
+---
+# Support public template
+inputs:
+  docfx.yml: |
+    template: "https://static.docs.com/ui/latest"
+  docs/a.yml: |
+    #YamlMime:TestData
+    test: b.md
+  docs/b.md:
+remoteFiles:
```
Comment (yufeih) [inline docs/specs/schema.yml:497]: nit: consider `http`?
```suggestion
http:
```

File: src/docfx/restore/PublicTemplatePackage.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        public override IEnumerable<PathString> GetFiles()
+        {
+            throw new NotSupportedException();
+        }
+
+        public override Stream ReadStream(PathString path) => _fileResolver.ReadStream(GetPath(path));
+
+        public override PathString? TryGetPhysicalPath(PathString path)
+        {
+            if (_fileResolver.TryResolveFilePath(GetPath(path), out var fullPath))
```
Comment (yufeih) [inline src/docfx/restore/PublicTemplatePackage.cs:35]: We should never hit this right?
```suggestion
            throw new NotSupportedException();
```

File: src/docfx/restore/PublicTemplatePackage.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        public PublicTemplatePackage(string baseUrl, FileResolver fileResolver)
+        {
+            _baseUrl = new Uri($"{baseUrl.TrimEnd('/')}/");
+            _fileResolver = fileResolver;
+        }
+
+        public override bool Exists(PathString path)
+        {
+            return _fileResolver.TryResolveFilePath(GetPath(path), out _);
```
Comment (yufeih) [inline src/docfx/restore/PublicTemplatePackage.cs:23]: We should never hit this right?
```suggestion
    throw new NotSupportedException();
```

Comment (928PJY) [reply src/docfx/restore/PublicTemplatePackage.cs:23]: it has been used when getting the schema

---
In reply to: [488341115](https://github.com/dotnet/docfx/pull/6550#discussion_r488341115) [](ancestors = 488341115)

File: src/docfx/config/TestQuirks.cs [hunk 5]
```diff
@@ -11,6 +11,8 @@ internal static class TestQuirks
 
         public static Func<string, string>? GitRemoteProxy { get; set; }
 
+        public static Func<string, string?>? RemoteFileProxy { get; set; }
```
Comment (yufeih) [inline src/docfx/config/TestQuirks.cs:14]:
```suggestion
        public static Func<string, string?>? HttpProxy { get; set; }
```

**Review by yufeih — APPROVED** [review]

**Pull request merged**